tachyon/zk/plonk/circuit/floor_planner/empty_space.h

```diff
@@ -0,0 +1,46 @@
+// Copyright 2020-2022 The Electric Coin Company
+// Copyright 2022 The Halo2 developers
+// Use of this source code is governed by a MIT/Apache-2.0 style license that
+// can be found in the LICENSE-MIT.halo2 and the LICENCE-APACHE.halo2
+// file.
+
+#ifndef TACHYON_ZK_PLONK_CIRCUIT_FLOOR_PLANNER_EMPTY_SPACE_H_
+#define TACHYON_ZK_PLONK_CIRCUIT_FLOOR_PLANNER_EMPTY_SPACE_H_
+
+#include <stddef.h>
+
+#include <optional>
+
+#include "tachyon/base/range.h"
+#include "tachyon/export.h"
+
+namespace tachyon::zk {
+
+// An area of empty space within a column.
+class TACHYON_EXPORT EmptySpace {
+ public:
+  EmptySpace(size_t start, std::optional<size_t> end)
```

Comment:
Didn't really think about it at first, but after reading https://quuxplusone.github.io/blog/2021/11/09/pass-string-view-by-value/, I think it's good to copy it since the size is small!